activerecord/lib/active_record/enum.rb

```diff
@@ -169,7 +169,12 @@ def enum(definitions)
 
         _enum_methods_module.module_eval do
           pairs = values.respond_to?(:each_pair) ? values.each_pair : values.each_with_index
+          value_checked = {}
           pairs.each do |label, value|
+            # check a value exist in both key
+            raise ArgumentError, "value #{value} exist in both of #{value_checked[value]}, #{label}" if value_checked[value]
```

Comment:
Can you change the error message to say "Values must be unique. The value '#{value}' was used on #{valid_checked[value]} and #{label}.".